### ISSUE TITLE
Rule request: redundant type unions

### ISSUE THREAD

**Issue by cake-monotone** [astral-sh/ruff#12625]
Hi! I noticed that Ruff doesn't warn about redundant type unions.

```py
c: str | str = "example"
```

While this example is straightforward, it can become confusing  when there are more than two or three types involved.

**Comment by AlexWaygood** [comment]
Hiya! Have you tried enabling PYI016?

**Comment by cake-monotone** [comment]
OMG 😭. Thanks so much!!
I got missed
